[docs] Iterating on recent Joy UI Component page updates

samuelsycamore opened the pull request:
Part of #33998 

Following #34858, #35152, and #34838, as well as some internal discussions with the Core team about these docs, I wanted to do a quick second pass on these three pages (Alert, Aspect Ratio, and Avatar) together to apply some structural changes based on what we've discovered so far.

Changes include:
- removing info about overriding props (unnecessary on component pages, and soon we'll have a guide with #34990) 
- moving Anatomy section to the bottom of the page - I think this is preferable for now, though we've discussed creating a new tab on the page for this info
- rearranging the Customization section - common props like Variants, Sizes, Decorators, Colors, etc. get bumped to the top, and in a consistent order
- adding h3s under Basics when Customization h3s feel more... "basic" 😅 than "customization" (i.e. adding images to Avatars feels like a basic implementation of the component)
- standardizing "Usage with" sections as differentiated from Customization
- standardizing shared/repeated text
- adding a tiny adjustment to one of the Alert demos that never got committed before that PR was merged, which helps to make them look less like buttons

I had intended to include these edits with my revision of the Autocomplete page, but because that doc is so huge, it really needs its own PR. It's shown me some structural things we need to account for as these pages get bigger—namely, that if everything is classified as "Customization," then this header starts to lose meaning. We also need to be intentional about the order in which information is presented under the Customization header, to deliver a more consistent experience across the docs.

oliviertassinari commented on docs/data/joy/components/alert/AlertColors.tsx:
```diff
@@ -20,7 +20,7 @@ export default function AlertColors() {
         sx={{
           display: 'grid',
-          gridTemplateColumns: 'repeat(auto-fit, minmax(150px, 1fr))',
+          gridTemplateColumns: 'repeat(auto-fit, minmax(150px, 400px))',
```
Are you sure?

samuelsycamore replied:
I thought I was! 😅 I'm stumped. What's a better way to do it?

oliviertassinari replied:
I feel. that we don't need CSS Grid in this case, a <Stack> would do the job.

siriwatknp replied:
true.

samuelsycamore replied:
Ah that makes sense. The CSS Grid does feel a bit overcomplicated here. I'll try refactoring it.

samuelsycamore commented:
@oliviertassinari Moving the conversation from #35132—

1. Should we remove the **Usage** section entirely from these docs? Is it unnecessary/unhelpful?
2. **Usage with** sections - should we edit these headers to simply name the thing that the component is being used with (Avatar Group, Next.js Image component, etc.)?

oliviertassinari commented:
On 1. I think that it should be kept when we can't have a basic demo that is simple enough to be expanded by default, e.g. most of MUI Base: https://mui.com/base/react-switch/#basics

siriwatknp commented:
1. I think that the current `## Usage` is not helpful or incomplete. What I get to see is how to import it but it does not work when I copy the code to my project (I don't see anything on my screen):
   ```js
   import Avatar from '@mui/joy/Avatar';
    
    export default function MyApp() {
      return <Avatar />;
    }
   ```
   I think it can be removed, at least for Joy UI. I guess developers prefer to look at the demo and open a Sandbox if they want to make changes.
   
   Maybe changing to `## Import` sounds better to me because I can copy the import directly.
2. I think **Usage with** helps me understand that it involves other components. I feel that it is better than the previous version. Maybe just **With ...** is enough and using the component name sounds better to me (without the space):
   - `## With AvatarGroup`
   - `## With Badge`
   - `## With Next.js Image`
3. I don't see that much difference at this point.

samuelsycamore commented:
OK, I'm cool with removing the Usage section altogether, and ensuring the the Basics section includes a very basic demo that can serve as a more useful replacement for the what the Usage section attempts to communicate. In situations where the Usage section might provide details that aren't covered elsewhere, I'll try moving them to Basics or the Introduction.

samuelsycamore commented:
After reflecting on @siriwatknp's comment above, I agree that the import—specifically, being able to copy and paste it—is the most useful part of the Usage section code snippet. I moved this to the Basics section. I also added imports for the "Usage with" sections that involve other Joy UI components, so those can be easily copied and pasted in as well. I think it also helps to quickly visually clarify that this section involves introducing a new component.

siriwatknp commented:
🤩 Love it!

<img width="840" alt="Screen Shot 2565-11-17 at 11 56 30" src="https://user-images.githubusercontent.com/18292247/202359783-51d5720e-8876-4e4a-ba38-75c2a38fa940.png">

samuelsycamore commented:
Thanks for the clean new Alert demo @oliviertassinari! This PR should be ready to go, and now I think we have a much better template to proceed with.

siriwatknp reviewed (approved):
👍 Well done!
